hooks/pre_prompt.py

```diff
@@ -0,0 +1,16 @@
+import warnings
+
+try:
+    from ccds import __version__
+except ImportError:
+    __version__ = None  # ccds is not installed
```

Comment:
```suggestion
    importlib.metadata import version, PackageNotFoundError
    try:
        __version__ = version("cookiecutter-data-science")  # ccds is not installed
    except importlib.metadata.PackageNotFoundError:
        __version__ = "1.0.0"  # or None
```

I wanted to see if I could get this warning to show up so I:

- Install `ccds` `pip install cookiecutter-data-science==2.0.0`
- Create new project using the template from this branch `ccds -c 425-version-bugfix`
- No warning shows up!

In hindsight that makes sense, bc:
- `ccds.__version__` is not defined prior to this commit (`from ccds import __version__` will fail with `ImportError` on `cookiecutter-data-science==2.0.0`)
- In the hook, the `except` block sets `__version__ = None`
- `if __version__ is not None` prevents warning from displaying

But we want this warning to show up if someone is using `2.0.0`. One way to do that would be to use `importlib.metdata.version` in the hook itself (in the suggestion). Alternatively, we can remove the `if __version__ is not None` condition to display the warning whenever the `ccds.__version__` does not exist (all ccds versions prior to `2.0.1`).

Reply:
Thank you for catching this @r-b-g-b! When I was testing locally, it seemed like it was working - then I realized that I had an environment issue (combo of conda base env + pipx) that made it look as though it was doing the right thing when it was not.

I've added a fix that uses more specific import errors to suss out whether ccds is not installed (throws a `ModuleNotFoundError`) and whether it is installed but an older version (throws an `ImportError) to get the desired behavior. Can you confirm that this new fix works for you? I've uninstalled all the ccds libraries on my machine and progressively added things in the following way:

```bash
pip uninstall ccds
pipx uninstall ccds

git clone -b 425-version-bugfix git@github.com:drivendataorg/cookiecutter-data-science.git test-ccds
cd test-ccds
uv venv --python 3.10
uv pip install cookiecutter

# should *not* print the 2.0.1 warning, ccds not installed. will print cookiecutter deprecation message
cookiecutter . 
uv pip install cookiecutter-data-science

# *should* print the 2.0.1 warning, ccds 2.0.0 installed
uv run ccds .  
sed -i -e 's/2\.0\.0/2.0.1/' pyproject.toml
uv pip install -e .

# should *not* print the error, 2.0.1 installed
uv run ccds .  
```

Reply:
Heyyyy it works! I like the way you implemented it with `ModuleNotFoundError` / `ImportError`.